dotnet/src/Connectors/Connectors.Memory.Redis/RedisMemoryStore.cs

```diff
@@ -23,18 +23,71 @@ namespace Microsoft.SemanticKernel.Connectors.Memory.Redis;
 /// <remarks>The embedded data is saved to the Redis server database specified in the constructor.
 /// Similarity search capability is provided through the RediSearch module. Use RediSearch's "Index" to implement "Collection".
 /// </remarks>
-public sealed class RedisMemoryStore : IMemoryStore
+public class RedisMemoryStore : IMemoryStore, IDisposable
 {
     /// <summary>
     /// Create a new instance of semantic memory using Redis.
     /// </summary>
-    /// <param name="database">The database of the redis server.</param>
-    /// <param name="vectorSize">Embedding vector size</param>
-    public RedisMemoryStore(IDatabase database, int vectorSize)
+    /// <param name="database">The database of the Redis server.</param>
+    /// <param name="vectorSize">Embedding vector size, defaults to 1536</param>
+    /// <param name="vectorIndexAlgorithm">Indexing algorithm for vectors, defaults to "HNSW"</param>
+    /// <param name="vectorDistanceMetric">Metric for measuring vector distances, defaults to "COSINE"</param>
+    /// <param name="queryDialect">Query dialect, must be 2 or greater for vector similarity searching, defaults to 2</param>
+    public RedisMemoryStore(
+        IDatabase database,
+        int vectorSize = VectorSize,
+        VectorIndexAlgorithms vectorIndexAlgorithm = VectorIndexAlgorithm,
+        VectorDistanceMetrics vectorDistanceMetric = VectorDistanceMetric,
+        int queryDialect = QueryDialect)
     {
+        if (vectorSize <= 0)
+        {
+            throw new ArgumentException("Vector size must be a positive integer.");
+        }
+
+        if (!Enum.TryParse<Schema.VectorField.VectorAlgo>(vectorIndexAlgorithm.ToString(), out this._vectorIndexAlgorithm))
+        {
+            throw new ArgumentException("Unsupported vector indexing algorithm.");
+        }
+
         this._database = database;
         this._vectorSize = vectorSize;
         this._ft = database.FT();
+        this._vectorDistanceMetric = vectorDistanceMetric.ToString();
```

Comment:
What is the likelihood of a new vector distance metric being defined?

When that happens our code needs to change whereas if we just take a string we are future proof but lose some strictness. Which is the lesser evil?

Reply:
Ideal scenario is when Redis SDK provides enum for this parameter (like for `vectorIndexAlgorithm` for example), and we just re-use it. As soon as new distance metric is added to enum, we just update Redis SDK package version. 

As for now, I would choose more secure scenario, when we provide enum on our side, as we know that these are supported values and we decrease the probability of error. I don't think supported values will change a lot, so the cost of maintenance is low.